sqlmesh/core/snapshot/definition.py

```diff
@@ -1258,40 +1255,44 @@ def missing_intervals(
 
 
 def earliest_start_date(
-    snapshots: t.Iterable[Snapshot],
+    snapshots: t.Collection[Snapshot],
     cache: t.Optional[t.Dict[str, datetime]] = None,
-    default_value: t.Optional[TimeLike] = None,
+    relative_to: t.Optional[TimeLike] = None,
 ) -> datetime:
     """Get the earliest start date from a collection of snapshots.
 
     Args:
         snapshots: Snapshots to find earliest start date.
         cache: optional cache to make computing cache date more efficient
+        relative_to: the base date to compute start from if inferred from cron
     Returns:
         The earliest start date or yesterday if none is found.
     """
     cache = {} if cache is None else cache
-    snapshots = list(snapshots)
-    earliest = to_datetime(default_value or yesterday().date())
     if snapshots:
-        return min(start_date(snapshot, snapshots, cache) or earliest for snapshot in snapshots)
-    return earliest
+        return min(
+            start_date(snapshot, snapshots, cache=cache, relative_to=relative_to)
+            for snapshot in snapshots
+        )
+    return yesterday()
```

Comment:
[Nit] Should we just raise if snapshots are empty?

Reply:
thought about it and think it may be dangerous